### DISCUSSION TITLE
Saving on close/reload (web)

### DISCUSSION THREAD

**Discussion by Mathmagician8191** [emilk/egui#2299] [Q&A]
I have a GUI that saves some configuration parameters. When running natively, the state is successfully saved and loaded, but when using trunk to make a web version, the state doesn't save when the tab is closed or reloaded immediately after changing the values, but does save if left open for a while (I assume that means autosave is working).

Is there a way to make it save when the tab is closed or reloaded?

**Comment by vaaandark** [comment]
I encountered the same problem. Have you solved it now?

**Reply by stonerfish** [reply]
If you look inside the [sources](https://github.com/emilk/egui/blob/master/crates/eframe/src/epi/mod.rs) you will see the definition of the "App" trait.  There are a few methods you can override.  One of them is auto_save_interval.
```
    /// Time between automatic calls to [`Self::save`]
    fn auto_save_interval(&self) -> std::time::Duration {
        std::time::Duration::from_secs(30)
    }
```

**Reply by vaaandark** [reply]
Thank you but I am afraid that this won't work for me. I don't think I should set the interval to a quite small number to save data on closing or reloading. It will cost too much expense!
I think egui can provide a interface to override the event to be done when reloading.
